pkg/server/server.go

```diff
@@ -17,8 +21,36 @@ func NewEkadashiServer(db *mongo.Service) (*EkadashiServer, error) {
 		Router: mux.NewRouter(),
 		db:     db,
 	}
-	s.Methods("POST").Path("/register").HandlerFunc(s.handleRegistration)
-	s.Methods("POST").Path("/login").HandlerFunc(s.handleLogin)
-	s.Methods("GET").Path("/users").HandlerFunc(s.showAllUsers)
+	s.Router.Handle("/register", chainMiddleware(http.HandlerFunc(s.handleRegistration), withLogging, withTracing)).
+		Methods("POST")
+	s.Router.Handle("/login", chainMiddleware(http.HandlerFunc(s.handleLogin), withLogging, withTracing)).
+		Methods("POST")
+	s.Router.Handle("/users", chainMiddleware(http.HandlerFunc(s.showAllUsers), withTracing, withLogging)).
+		Methods("GET")
 	return s, nil
 }
+
+type middleware func(http.Handler) http.Handler
+
+func chainMiddleware(handler http.Handler, mw ...middleware) http.Handler {
+	for _, m := range mw {
+		handler = m(handler)
+	}
+	return handler
+}
+
+func withLogging(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Printf("Logged connection from %s", r.RemoteAddr)
```

Comment:
Logger should print request and response, method, status code, URL, also tracing can be done here